src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/TitleController.php

```diff
@@ -0,0 +1,155 @@
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Controller responsible of "Configure > Shop Parameters > Customer Settings > Titles" page.
+ */
+class TitleController extends FrameworkBundleAdminController
+{
+    /**
+     * Show customer titles page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, TitleFilters $filters): Response
+    {
+        $titleGridFactory = $this->get('prestashop.core.grid.factory.title');
+        $titleGrid = $titleGridFactory->getGrid($filters);
+
+        return $this->render('@PrestaShop/Admin/Configure/ShopParameters/CustomerSettings/Title/index.html.twig', [
+            'titleGrid' => $this->presentGrid($titleGrid),
+            'layoutTitle' => $this->trans('Titles', 'Admin.Navigation.Menu'),
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+        ]);
+    }
+
+    /**
+     * Displays and handles currency form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_title_index",
+     *     message="You need permission to create this."
+     * )
+     *
+     * @return Response
+     */
+    public function createAction(): Response
+    {
+        return $this->redirect(
+            $this->getContext()->link->getAdminLink(
+                'AdminGenders',
+                true,
+                [],
+                [
+                    'addgender' => '',
+                ]
+            )
+        );
+    }
+
+    /**
+     * Displays title form.
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_title_index",
+     *     message="You need permission to edit this."
+     * )
+     *
+     * @return Response
+     */
+    public function editAction(int $titleId): Response
+    {
+        return $this->redirect(
+            $this->getContext()->link->getAdminLink(
+                'AdminGenders',
+                true,
+                [],
+                [
+                    'updategender' => '',
+                    'id_gender' => $titleId,
+                ]
+            )
+        );
+    }
+
+    /**
+     * Deletes title.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_title_index",
+     *     message="You need permission to delete this."
+     * )
+     * @DemoRestricted(redirectRoute="admin_title_index")
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction(int $titleId): RedirectResponse
+    {
+        return $this->redirect(
```

Comment:
$this->redirectToRoute('routename'); is not applicable ?

Reply:
Not because I haven't migrated at the moment the delete action (and all actions).